### PR TITLE
ARROW-12550: [Website] Write blog post describing new Rust dev processes

### PR DESCRIPTION
**Pull request by returnString**
Putting this up in its current state for initial review and feedback from the community.

### CODE REVIEW

File: _posts/2021-04-27-rust-dev-workflow.md
Comment (fitzoh) [inline _posts/2021-04-27-rust-dev-workflow.md:38]: `ASF` is used a couple times, worth defining?

File: _posts/2021-04-27-rust-dev-workflow.md
Comment (fitzoh) [inline _posts/2021-04-27-rust-dev-workflow.md:45]: I was unaware of Spark compatibility and couldn't find that claim on a very cursory search.
Is it compatible with Spark or comparable to Spark?

Comment (returnString) [reply _posts/2021-04-27-rust-dev-workflow.md:45]: It seems I imagined this - can't see it in any of the reference material I was looking at yesterday. Will remove!

File: _posts/2021-04-27-rust-dev-workflow.md
Comment (Dandandan) [inline _posts/2021-04-27-rust-dev-workflow.md:59]:
```suggestion
To further improve the onboarding flow for new Arrow contributors, we have started the process of labeling select issues as "good first issue" in [arrow-rs][11] and [arrow-datafusion][12]. These issues are small in scope but still serve as valuable contributions to the project. They also help new community members to get familiar with our development workflows and tools.
```
Some stylistic suggestions

Comment (returnString) [reply _posts/2021-04-27-rust-dev-workflow.md:59]: Ah, the old single vs double L, a classic in English dialect wars! Lemme scan through the blog post history, I'll standardise (heh, I'm picking a side already with that S) on either British or American English depending on which is more common. There's probably other bits that need unifying if so.

I'll try and incorporate these suggestions in some fashion, clarity is always a good goal 👍

**Review by andygrove — APPROVED** [review]
This looks great! Thanks @returnString

File: _posts/2021-04-27-rust-dev-workflow.md
Comment (alamb) [inline _posts/2021-04-27-rust-dev-workflow.md:30]:
```suggestion
- Enables a faster release cadence and versioning scheme that better matches Cargo's SemVer expectations
```

File: _posts/2021-04-27-rust-dev-workflow.md
Comment (alamb) [inline _posts/2021-04-27-rust-dev-workflow.md:31]:
```suggestion
- Encourages maximum participation from the wider community by using only Github (rather than both Github and JIRA)
```

**Review by alamb — APPROVED** [review]
I think it looks great. Thank you @returnString !

I left some suggestions that I am not 100% thrilled with -- what I was trying to do was to highlight the specific changes that we hope to make (namely using crate versioning more in line with semver, as well as centralizing on a single tool)

However, it makes the initial bullet points more verbose. I do not feel strongly about this.

All in all, this is really nice

File: _posts/2021-04-27-rust-dev-workflow.md
Comment (returnString) [reply _posts/2021-04-27-rust-dev-workflow.md]: > I think it looks great. Thank you @returnString !
> 
> I left some suggestions that I am not 100% thrilled with -- what I was trying to do was to highlight the specific changes that we hope to make (namely using crate versioning more in line with semver, as well as centralizing on a single tool)
> 
> However, it makes the initial bullet points more verbose. I do not feel strongly about this.
> 
> All in all, this is really nice

I've taken the spirit of these suggestions and tried to incorporate them without affecting the brevity of the bullet points too significantly - SemVer was worth mentioning for sure!

**Comment by returnString** [comment]
I'm happy with this initial round of feedback, so promoting this from draft to ready-for-review - thanks all for your effort so far 😄

File: _posts/2021-04-27-rust-dev-workflow.md
Comment (fitzoh) [inline _posts/2021-04-27-rust-dev-workflow.md:40]: Possibly worth mentioning that this is a shift back towards the process used by Ballista?

Adds the context of "going back to a thing that worked" rather than "let's try something new".

File: _posts/2021-04-27-rust-dev-workflow.md
Comment (jorgecarleitao) [inline _posts/2021-04-27-rust-dev-workflow.md:45]: We have more than 3 crates; within `arrow-rs` we have 4 or 5. This split is more related to the nature of these crates (AFAIK).

**Review by jorgecarleitao — APPROVED** [review]
Looks great. I left some minor comments for consideration.

File: _posts/2021-04-27-rust-dev-workflow.md
Comment (returnString) [reply _posts/2021-04-27-rust-dev-workflow.md:35]: I was trying to avoid repeating the word repository too often - difficult in a post on dev and release workflows, admittedly 😅

**Comment by alamb** [comment]
@andygrove  is this one good to go? I think it is ready

**Comment by returnString** [comment]
Not sure what the policy is on publish dates but if it needs updating to a specific date before merging then please let me know and I can patch that up to today's date; alternatively, maintainer access is enabled for my branch, so PMC members should be able to fix that directly before merging.

**Comment by andygrove** [comment]
LGTM. I will share on the dev@ mailing list as a courtesy as well.

File: _posts/2021-04-27-rust-dev-workflow.md
Comment (emkornfield) [inline _posts/2021-04-27-rust-dev-workflow.md:57]: It might help to have language that opening issues before PRs are submitted is recommended to let others know you are working on it and discuss technical issues before work is put into a PR.  i.e. this is more a tooling change then anything else.

Comment (andygrove) [reply _posts/2021-04-27-rust-dev-workflow.md:57]: @returnString If you could address this point and update the publish date I think we can go ahead and get this published.

Comment (returnString) [reply _posts/2021-04-27-rust-dev-workflow.md]: Thanks for the feedback @emkornfield! I've incorporated it into that section.

@andygrove I think we're good to go 😄

**Review by andygrove — APPROVED** [review]

**Comment by alamb** [comment]
🎉  Thanks again @returnString  everyone else who contributed to this post

**Pull request merged**
